### PR TITLE
First release

### PR DESCRIPTION
**Pull request by arnaudbore**

### CODE REVIEW

File: USAGE [hunk 1]
```diff
@@ -0,0 +1,54 @@
+toTemplate_flow pipeline
```
Comment (CHrlS98) [inline USAGE]: `toTemplate_flow` or `register_flow`?

File: USAGE [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+OPTIONAL ARGUMENTS (current value)
+    --linear                                    If set, it will use linear registration instead of non linear registration.
+
+    --slow_registration                         If set, it will use antsRegistrationSyN instead of antsRegistrationSyNQuick for registration.
+
+    --output_dir                                Specify a new output directory to avoid overwriting results (when dealing with multiple input labels) ($output_dir)
+
+    --processes_register                        Number of processes for registration task ($processes_register)
+
+AVAILABLE PROFILES (using -profile option (e.g. -profile use_cuda,fully_reproducible))
```
Comment (CHrlS98) [inline USAGE]: Is `use_cuda` an available profile ? I get that it is an example but maybe we should put an available profile for example?

Comment (CHrlS98) [inline USAGE]: Is `use_cuda` an available profile ? I get that it is an example but maybe we should put an available profile for example?

File: USAGE [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+                                                ├── S1
+                                                │   │── *t1.nii.gz (diff space)
+                                                |   └── metrics (optional)
+                                                |   │   └── *.nii.gz
+                                                |   └── tractograms (optional)
+                                                |       └── *.trks
+                                                └── S2
+                                                    └── *
+
+    --template                                  Template used for registration.
```
Comment (CHrlS98) [inline USAGE]: I guess a T1 is best, but are there other images that would work? Should we write that a T1, or at least that a nifti file, is expected?

Comment (CHrlS98) [inline USAGE]: I guess a T1 is best, but are there other images that would work? Should we write that a T1 is expected, or at least that a nifti file is expected?

**Review by CHrlS98 — COMMENTED** [review]
I tried with the default parameters and the optional ones after running `combine_flows` on a `tractoflow` output.
Works really well. Only thing I noticed is that the registered nufo has floating point values. Is this expected?

**Review by CHrlS98 — APPROVED** [review]

File: USAGE [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+                                                |   │   └── *.nii.gz
+                                                |   └── tractograms (optional)
+                                                |       └── *.trks
+                                                └── S2
+                                                    └── *
+
+    --template=/path/to/[template]              Path to a nifti T1 template used for registration.
+
+OPTIONAL ARGUMENTS (current value)
+    --linear                                    If set, it will use linear registration instead of non linear registration.
```
Comment (mdesco) [inline USAGE]: Please add defaults in ()

File: USAGE [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+    --slow_registration                         If set, it will use antsRegistrationSyN instead of antsRegistrationSyNQuick for registration.
```
Comment (mdesco) [inline USAGE]: add default in ().
Also, please add the slow option as default and rename this option --quick_registration

I think we should always use slow registration unless we are prototyping

**Review by mdesco — CHANGES_REQUESTED** [review]

File: USAGE (hunk 5 above)
Comment (mdesco) [reply USAGE]: Just tested and the --slow_registration option went super fast on Cipaye on 12 subjects. Are you sure that you are truly activating the non-quick version? :)

**Review by mdesco — APPROVED** [review]
Looks good. @CHrlS98 is about to try to it. We can raise issue

File: main.nf [hunk 6]
```diff
@@ -0,0 +1,217 @@
...
+
+log.info "Number of processes per tasks"
+log.info "============================="
+log.info "Template registration: $params.processes_register"
+log.info ""
+
+root = file(params.input)
+/* Watch out, files are ordered alphabetically in channel */
+Channel
+    .fromPath("$root/**/t1.nii.gz",
```
Comment (CHrlS98) [inline main.nf]: There should be an `*` here before `t1.nii.gz` in order to deal properly with multiple subjects.

**Review by CHrlS98 — COMMENTED** [review]
Once I fixed the `*` issue (see my comment), it worked perfectly using default parameters. I think it's safe to merge once it is fixed. We'll raise issues if we encounter problems + I'll be able to do PR to fix stuff.

**Review by CHrlS98 — APPROVED** [review]
Tried with default options, then with `--quick_registration` and `--linear_registration`. Everything seems to be working perfectly! Ready to merge!

**Pull request merged**